Data Acquisition
There are various formats for a dataset such as .csv, .json, .xlsx etc. The dataset can be stored in different places on your local machine or sometimes online.
In our case, the Automobile Dataset is a local source and it is in CSV (comma separated value) format.

data source: "C:\Users\GeekStore\Desktop\Data_Mining&Machine_Learning_Project\wandaloo_cars.csv"
data type: csv
The Pandas Library is a useful tool that enables us to read various datasets into a data frame; our Jupyter notebook platforms have a built-in Pandas Library so that all we need to do is import Pandas without installing.
Use pandas.read_csv() function to read the csv file. In the bracket, put the file path along so that pandas will read the file into a data frame from that address. The file path can be either an URL or local file address.
Because the data does not include headers, add an argument headers = None inside the read_csv() method, so that pandas will not automatically set the first row as a header.

In [2]:
import pandas as pd

df = pd.read_csv('wandaloo_cars.csv')
print("The first 5 rows of the dataframe") 
df.head(5)

The first 5 rows of the dataframe


Brand           Model        Version       Price  Modèle     Main  \
0  MERCEDES   Classe C 2014          220 d  228.000 DH  2014.0  Seconde   
1  MERCEDES   Classe C 2003        220 CDI   79.000 DH  2003.0  Seconde   
2  MERCEDES   Classe B 2013    180 d Urban  150.000 DH  2013.0  Seconde   
3  MERCEDES   Classe E 2006  Autre version  135.000 DH  2006.0  Seconde   
4  MERCEDES  CLA Coupé 2014        220 CDI  240.000 DH  2014.0  Seconde   

  Kilométrage Carburant  Transmision Puissance fiscale  ... Airbags  ABS  ESP  \
0  144.500 km    Diesel  Automatique              8 cv  ...     oui  oui  oui   
1  350.000 km    Diesel     Manuelle              8 cv  ...     oui  oui  oui   
2  144.000 km    Diesel     Manuelle              8 cv  ...     oui  oui  oui   
3  248.000 km    Diesel  Automatique              9 cv  ...     oui  oui  oui   
4  103.000 km    Diesel  Automatique              9 cv  ...     oui  oui  oui   

  Antipatinage Architecture Cylindrée Conso. ville Conso. route Vitesse maxi.  \
0          oui          NaN       NaN          NaN          NaN           NaN   
1          oui          NaN       NaN          NaN          NaN           NaN   
2          oui          NaN       NaN          NaN          NaN           NaN   
3          oui          NaN       NaN          NaN          NaN           NaN   
4          oui          NaN       NaN          NaN          NaN           NaN   

  Volume du réservoir  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4                 NaN  

[5 rows x 43 columns]

In [3]:
df.tail(10)

Brand Model Version                           Price  Modèle Main  \
3119  VOLVO  XC40     NaN    560.000 DH *   * Prix public     NaN  NaN   
3120  VOLVO  XC60     NaN    499.000 DH *   * Prix public     NaN  NaN   
3121  VOLVO  XC60     NaN    559.000 DH *   * Prix public     NaN  NaN   
3122  VOLVO  XC60     NaN    800.000 DH *   * Prix public     NaN  NaN   
3123  VOLVO  XC60     NaN    820.000 DH *   * Prix public     NaN  NaN   
3124  VOLVO  XC90     NaN    630.000 DH *   * Prix public     NaN  NaN   
3125  VOLVO  XC90     NaN    730.000 DH *   * Prix public     NaN  NaN   
3126  VOLVO  XC90     NaN    850.000 DH *   * Prix public     NaN  NaN   
3127  VOLVO  XC90     NaN  1.000.000 DH *   * Prix public     NaN  NaN   
3128  VOLVO  XC90     NaN  1.100.000 DH *   * Prix public     NaN  NaN   

     Kilométrage Carburant Transmision Puissance fiscale  ... Airbags  ABS  \
3119         NaN       NaN         NaN              8 cv  ...     7.0  oui   
3120         NaN       NaN         NaN              8 cv  ...     7.0  oui   
3121         NaN       NaN         NaN              8 cv  ...     7.0  oui   
3122         NaN       NaN         NaN             11 cv  ...     7.0  oui   
3123         NaN       NaN         NaN             11 cv  ...     7.0  oui   
3124         NaN       NaN         NaN              8 cv  ...     7.0  oui   
3125         NaN       NaN         NaN              8 cv  ...     7.0  oui   
3126         NaN       NaN         NaN              8 cv  ...     7.0  oui   
3127         NaN       NaN         NaN             11 cv  ...     7.0  oui   
3128         NaN       NaN         NaN             11 cv  ...     7.0  oui   

      ESP Antipatinage Architecture  Cylindrée  Conso. ville  Conso. route  \
3119  oui          oui  3 cylindres  1.477 cm³    - l/100 km    - l/100 km   
3120  oui          oui  4 cylindres  1.969 cm³  5,5 l/100 km  4,3 l/100 km   
3121  oui          oui  4 cylindres  1.969 cm³  5,5 l/100 km  4,3 l/100 km   
3122  oui          oui  4 cylindres  1.969 cm³    - l/100 km    - l/100 km   
3123  oui          oui  4 cylindres  1.969 cm³    - l/100 km    - l/100 km   
3124  oui          oui  4 cylindres  1.969 cm³  6,3 l/100 km  5,4 l/100 km   
3125  oui          oui  4 cylindres  1.969 cm³  6,3 l/100 km  5,4 l/100 km   
3126  oui          oui  4 cylindres  1.969 cm³  6,3 l/100 km  5,4 l/100 km   
3127  oui          oui  4 cylindres  1.969 cm³    - l/100 km    - l/100 km   
3128  oui          oui  4 cylindres  1.969 cm³    - l/100 km    - l/100 km   

     Vitesse maxi. Volume du réservoir  
3119      180 km/h            54 litre  
3120      210 km/h            70 litre  
3121      210 km/h            70 litre  
3122      180 km/h            70 litre  
3123      180 km/h            70 litre  
3124      220 km/h            71 litre  
3125      220 km/h            71 litre  
3126      220 km/h            71 litre  
3127      230 km/h            71 litre  
3128      230 km/h            71 litre  

[10 rows x 43 columns]

Drop the missing values along the column "price"

In [4]:
df.dropna(subset=["Price"], axis=0)

Brand           Model        Version                           Price  \
0     MERCEDES   Classe C 2014          220 d                      228.000 DH   
1     MERCEDES   Classe C 2003        220 CDI                       79.000 DH   
2     MERCEDES   Classe B 2013    180 d Urban                      150.000 DH   
3     MERCEDES   Classe E 2006  Autre version                      135.000 DH   
4     MERCEDES  CLA Coupé 2014        220 CDI                      240.000 DH   
...        ...             ...            ...                             ...   
3124     VOLVO            XC90            NaN    630.000 DH *   * Prix public   
3125     VOLVO            XC90            NaN    730.000 DH *   * Prix public   
3126     VOLVO            XC90            NaN    850.000 DH *   * Prix public   
3127     VOLVO            XC90            NaN  1.000.000 DH *   * Prix public   
3128     VOLVO            XC90            NaN  1.100.000 DH *   * Prix public   

      Modèle     Main Kilométrage Carburant  Transmision Puissance fiscale  \
0     2014.0  Seconde  144.500 km    Diesel  Automatique              8 cv   
1     2003.0  Seconde  350.000 km    Diesel     Manuelle              8 cv   
2     2013.0  Seconde  144.000 km    Diesel     Manuelle              8 cv   
3     2006.0  Seconde  248.000 km    Diesel  Automatique              9 cv   
4     2014.0  Seconde  103.000 km    Diesel  Automatique              9 cv   
...      ...      ...         ...       ...          ...               ...   
3124     NaN      NaN         NaN       NaN          NaN              8 cv   
3125     NaN      NaN         NaN       NaN          NaN              8 cv   
3126     NaN      NaN         NaN       NaN          NaN              8 cv   
3127     NaN      NaN         NaN       NaN          NaN             11 cv   
3128     NaN      NaN         NaN       NaN          NaN             11 cv   

      ... Airbags  ABS  ESP Antipatinage Architecture  Cylindrée  \
0     ...     oui  oui  oui          oui          NaN        NaN   
1     ...     oui  oui  oui          oui          NaN        NaN   
2     ...     oui  oui  oui          oui          NaN        NaN   
3     ...     oui  oui  oui          oui          NaN        NaN   
4     ...     oui  oui  oui          oui          NaN        NaN   
...   ...     ...  ...  ...          ...          ...        ...   
3124  ...     7.0  oui  oui          oui  4 cylindres  1.969 cm³   
3125  ...     7.0  oui  oui          oui  4 cylindres  1.969 cm³   
3126  ...     7.0  oui  oui          oui  4 cylindres  1.969 cm³   
3127  ...     7.0  oui  oui          oui  4 cylindres  1.969 cm³   
3128  ...     7.0  oui  oui          oui  4 cylindres  1.969 cm³   

      Conso. ville  Conso. route Vitesse maxi. Volume du réservoir  
0              NaN           NaN           NaN                 NaN  
1              NaN           NaN           NaN                 NaN  
2              NaN           NaN           NaN                 NaN  
3              NaN           NaN           NaN                 NaN  
4              NaN           NaN           NaN                 NaN  
...            ...           ...           ...                 ...  
3124  6,3 l/100 km  5,4 l/100 km      220 km/h            71 litre  
3125  6,3 l/100 km  5,4 l/100 km      220 km/h            71 litre  
3126  6,3 l/100 km  5,4 l/100 km      220 km/h            71 litre  
3127    - l/100 km    - l/100 km      230 km/h            71 litre  
3128    - l/100 km    - l/100 km      230 km/h            71 litre  

[3063 rows x 43 columns]

In [5]:
print("COLUMN NAMES")
print(df.columns)

COLUMN NAMES
Index(['Brand', 'Model', 'Version', 'Price', 'Modèle', 'Main', 'Kilométrage',
       'Carburant', 'Transmision', 'Puissance fiscale', 'Couleur extérieure',
       'Etat du véhicule', 'Climatisation', 'Vitres électriques',
       'Sièges électriques', 'Ordinateur de bord', 'Start & Stop',
       'Régulateur de vitesse', 'Allumage auto. des feux',
       'Détecteur de pluie', 'Commandes au volant', 'Ecran tactile',
       'Rétroviseurs électriques', 'Ouverture auto. du coffre',
       'Démarrage mains libres', 'Banquette arrière rabattable 1/3-2/3',
       'Caméra de recul', 'Bluetooth', 'Jantes aluminium', 'Volant cuir',
       'Feux de jour', 'Barres de toit', 'Toit', 'Airbags', 'ABS', 'ESP',
       'Antipatinage', 'Architecture', 'Cylindrée', 'Conso. ville',
       'Conso. route', 'Vitesse maxi.', 'Volume du réservoir'],
      dtype='object')


Data Types
Data has a variety of types.The main types stored in Pandas dataframes are object, float, int, bool and datetime64. In order to better learn about each attribute, it is always good to know the data type of each column. In Pandas:

In [6]:
df.dtypes

Brand                                    object
Model                                    object
Version                                  object
Price                                    object
Modèle                                  float64
Main                                     object
Kilométrage                              object
Carburant                                object
Transmision                              object
Puissance fiscale                        object
Couleur extérieure                       object
Etat du véhicule                         object
Climatisation                            object
Vitres électriques                       object
Sièges électriques                       object
Ordinateur de bord                       object
Start & Stop                             object
Régulateur de vitesse                    object
Allumage auto. des feux                  object
Détecteur de pluie                       object
Commandes au volant                     

It returns a Series with the data type of each column.

It is clear to see that the data type of "Main"and "Kilométrage are object, and "Modèle" is float64, etc.

In [7]:
df.describe()

Modèle
count  2180.000000
mean   2012.002294
std       5.407446
min    1984.000000
25%    2009.000000
50%    2013.000000
75%    2016.000000
max    2022.000000

This shows the statistical summary of all numeric-typed (int, float) columns.
For example, the attribute "Modèle" has 2180.000000 counts, the mean value of this column is 2012.002294, the standard deviation is 5.407446, the minimum value is 1984.000000, 25th percentile is 2009.000000, 50th percentile is 2013.000000, 75th percentile is 2016.000000, and the maximum value is 2022.000000.
However, To check all the columns including those that are of type object.


Add an argument include = "all" inside the bracket.

In [8]:
# describe all the columns in "df" 
df.describe(include = "all")

Brand Model        Version       Price       Modèle     Main  \
count         3063  3063           2180        3063  2180.000000     2180   
unique          33  1127            733        1111          NaN        2   
top     VOLKSWAGEN   GLC  Autre version  110.000 DH          NaN  Seconde   
freq           532    24            374          36          NaN     1126   
mean           NaN   NaN            NaN         NaN  2012.002294      NaN   
std            NaN   NaN            NaN         NaN     5.407446      NaN   
min            NaN   NaN            NaN         NaN  1984.000000      NaN   
25%            NaN   NaN            NaN         NaN  2009.000000      NaN   
50%            NaN   NaN            NaN         NaN  2013.000000      NaN   
75%            NaN   NaN            NaN         NaN  2016.000000      NaN   
max            NaN   NaN            NaN         NaN  2022.000000      NaN   

       Kilométrage Carburant Transmision Puissance fiscale  ... Airbags   ABS  \
count         2180      2166        2180              3048  ...    3048  3048   
unique         632         4           2                42  ...      11     2   
top     200.000 km    Diesel    Manuelle              6 cv  ...     oui   oui   
freq            47      1902        1146               955  ...    1922  2704   
mean           NaN       NaN         NaN               NaN  ...     NaN   NaN   
std            NaN       NaN         NaN               NaN  ...     NaN   NaN   
min            NaN       NaN         NaN               NaN  ...     NaN   NaN   
25%            NaN       NaN         NaN               NaN  ...     NaN   NaN   
50%            NaN       NaN         NaN               NaN  ...     NaN   NaN   
75%            NaN       NaN         NaN               NaN  ...     NaN   NaN   
max            NaN       NaN         NaN               NaN  ...     NaN   NaN   

         ESP Antipatinage Architecture  Cylindrée Conso. ville Conso. route  \
count   3048         3048          845        845          845          845   
unique     2            2           11         69           85           62   
top      oui          oui  4 cylindres  1.968 cm³   - l/100 km   - l/100 km   
freq    2235         2094          549         84          121          121   
mean     NaN          NaN          NaN        NaN          NaN          NaN   
std      NaN          NaN          NaN        NaN          NaN          NaN   
min      NaN          NaN          NaN        NaN          NaN          NaN   
25%      NaN          NaN          NaN        NaN          NaN          NaN   
50%      NaN          NaN          NaN        NaN          NaN          NaN   
75%      NaN          NaN          NaN        NaN          NaN          NaN   
max      NaN          NaN          NaN        NaN          NaN          NaN   

       Vitesse maxi. Volume du réservoir  
count            868                 845  
unique            98                  51  
top         250 km/h            50 litre  
freq              78                 144  
mean             NaN                 NaN  
std              NaN                 NaN  
min              NaN                 NaN  
25%              NaN                 NaN  
50%              NaN                 NaN  
75%              NaN                 NaN  
max              NaN                 NaN  

[11 rows x 43 columns]

Now, it provides the statistical summary of all the columns, including object-typed attributes.
We can now see how many unique values, which is the top value and the frequency of top value in the object-typed columns.
Some values in the table above show as "NaN", this is because those numbers are not available regarding a particular column type.

In [10]:
df.info

<bound method DataFrame.info of          Brand           Model        Version                           Price  \
0     MERCEDES   Classe C 2014          220 d                      228.000 DH   
1     MERCEDES   Classe C 2003        220 CDI                       79.000 DH   
2     MERCEDES   Classe B 2013    180 d Urban                      150.000 DH   
3     MERCEDES   Classe E 2006  Autre version                      135.000 DH   
4     MERCEDES  CLA Coupé 2014        220 CDI                      240.000 DH   
...        ...             ...            ...                             ...   
3124     VOLVO            XC90            NaN    630.000 DH *   * Prix public   
3125     VOLVO            XC90            NaN    730.000 DH *   * Prix public   
3126     VOLVO            XC90            NaN    850.000 DH *   * Prix public   
3127     VOLVO            XC90            NaN  1.000.000 DH *   * Prix public   
3128     VOLVO            XC90            NaN  1.100.000 DH *   * Prix public

It shows us that the whole data frame has 3129 rows and 43 columns in total.